### ISSUE TITLE
Max val_size

### ISSUE THREAD

**Issue by rnyak** [IndicoDataSolutions/finetune#428]
**Describe the bug**
Is max val_size calculated from the given formula:
 "If n_examples > 50, defaults to max(5, min(100, 0.05 * n_examples))"?

In other words, if I have a training set of 30k samples, can I set the val_size = 3000 or 5000? When I do that, I do not see training and test set loss values on TensorBoard. There is nothing on the loss plot.

**Minimum Reproducible Example**
A short code snippet which reproduces the exception

**Expected behavior**

**Additional context**

**Comment by benleetownsend** [comment]
You should also set `val_interval` to choose how frequently you would like to run a validation set.

**Comment by rnyak** [comment]
Thanks, I will try and see the results.

**Comment by madisonmay** [comment]
@rnyak did that do the trick?  The culprit here is a heuristic we apply to help ensure running validation doesn't slow your model down by more than a percentage we deem acceptable (I believe 50%, in this case).  So the heuristic for val_interval may mean that validation almost never gets run if you specify a large validation set.

**Comment by rnyak** [comment]
I see. Thanks @madisonmay. What I want is to validate my model with a val set with size of 10% of the samples within one step, and use val loss as earl_stopping criteria.
I was thinking to use`val_set: Where it is necessary to use an explicit validation set, provide it here as a tuple (text, labels)` parameter. I need to test which param would work better for my case: val_size or val_set.

**Comment by rnyak** [comment]
@madisonmay @benleetownsend , when we set val_size = 100 (as an example), apparently 100 samples out of the training samples become val_set. Are these 100 samples randomly sampled from training samples?  also, does val set changes at each epoch ( i.e., resampled at each epoch), or it is fixed?

**Comment by benleetownsend** [comment]
Val set is consistent at each epoch and are randomly sampled from the training set.
